internal/receiver/databricksreceiver/paginator.go

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package databricksreceiver
+
+import "fmt"
+
+// paginator is extracted from apiPaginator for swapping out in unit tests
+type paginator interface {
```

Comment:
nit, I think `paginator` is a bit of an implementation detail and wonder if `databricksClient` is preferable considering it's used by the metrics providers.

Reply:
Yeah, I see what you're saying. The thing is that if we call it databricksClient, I would probably get confused between it, apiClient, and http.Client. I kind of like paginator because it tells you exactly what it does :)

Reply:
I had similar thoughts where authClient < apiClient < paginator and "paginator" doesn't seem to reflect its role as the actual client handle. I'd think it's presumed a databricks client would correctly deal w/ paginated results.

Reply:
Makes sense, fixed. Wasn't sure what to name the interfaces so I just appended `Interface` which at least makes things pretty clear 😄

Reply:
Thanks, this is much easier for me to follow.